web/film-club-client/src/components/FilmDetailPage.js

```diff
@@ -1,54 +1,193 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { useParams } from 'react-router-dom';
-import { Box, Image, Heading, Text } from 'grommet';
+import { Box, Image, Heading, Layer, Text } from 'grommet';
+import { CirclePlay } from 'grommet-icons';
 
 import Header from './Header';
 
-const FilmDetailPage = props => {
+const FilmDetailPage = (props) => {
+	const [movieDetails, setMovieDetails] = useState({});
+	const [showVideo, setShowVideo] = useState(false);
+
 	// React Router URL Parameter object
 	let { id } = useParams();
 
-	// const getMovieDetails = async (id) => {
-	//     const response = await fetch(`http://localhost:8080/v1/films/${id}`)
-	//     const data = await response
-	//     // props.setMovieId(data)
-	//     console.log(data)
-	// }
-
-	// getMovieDetails(id)
+	// // Fetch all movie details with {id} from params, on component load
+	useEffect(() => {
+		// Use the {id} to fetch details about the corresponding movie
+		const getMovieDetails = async (id) => {
+			const response = await fetch(`http://localhost:8080/v1/films/${id}`);
+			const data = await response.json();
+			console.log(data);
+			setMovieDetails(data);
+		};
+		getMovieDetails(id);
+	}, [id]);
 
-	let { posterPath, title, overview } = props.movieId;
+	let {
+		backdropPath,
+		credits,
+		images,
+		genres,
+		posterPath,
+		releaseDate,
+		similar,
+		tagline,
+		title,
+		overview,
+		videos,
+		watchProviders,
+	} = movieDetails;
 
-	return (
-		<Box>
-			<Header />
-			{id ? (
+	if (movieDetails.images) {
```

Comment:
You're already destructuring `images`, don't need to access `images` from the `movieDetails`.